.github/workflows/build.yaml

```diff
@@ -16,7 +16,6 @@ jobs:
         with:
           # Needed for commands that depend on git tags
           fetch-depth: 0
-          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Why is this removed here but not for the other platform builds?

Reply:
Good catch. It isn't a big issue to keep/remove it, but it is just an unnecessary value,so I've now removed it from all other platforms.

Reply:
Hm I dug up this old PR https://github.com/livepeer/go-livepeer/pull/1891 and it seems that specifying `github.ref` is actually important because otherwise the checkout action will use the SHA of the merge commit that is created by merging the base to the head, but that merge commit might not be present in commit history (i.e. if a rebase/squash + merge strategy is used). Since the go-livepeer version string uses that SHA, if the merge commit does not exist in the commit history it could make it harder to debug builds.

Reply:
thanks for bringing that up. i think I'll need to update this across other projects using sha for ephemeral builds too.

added a comment with the pr link for future reference